libclamav/regex_suffix.c

```diff
@@ -274,7 +274,8 @@ static struct node *parse_regex(const uint8_t *p, const size_t pSize, size_t *la
     struct node *right;
     struct node *tmp;
 
-    while (p[*last] != '$' && p[*last] != '\0') {
+    while (*last < pSize && p[*last] != '$' && p[*last] != '\0') {
+        if (*last >= pSize) break;  // Safe guard
```

Comment:
I don't think this safe guard is needed. The check in the `while()`-condition does the same thing at a better place.
```suggestion
```

Reply:
Hii @val-ms Thanks for reviewing i had commited above suggested changes
Thanks